### PR TITLE
Ordering in data

### PR DESCRIPTION
**Pull request by IKupriyanov-HORIS**

### CODE REVIEW

File: plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/data/DataProcessing.kt [hunk 1]
```diff
@@ -151,6 +149,105 @@ object DataProcessing {
...
+
+        fun initOrderSpecs(
+            orderOptions: List<OrderOptionUtil.OrderOption>,
+            variables: Set<Variable>,
+            bindings: List<VarBinding>
+        ) {
+            if (myOrderSpecs != null) return
+            myOrderSpecs = orderOptions.filter { orderOption ->
+                // Skip positionals aes
+                val aes = Aes.values().find { it.name == orderOption.aesName }
```
Comment (alshan) [inline plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/data/DataProcessing.kt:164]: 'Ordering option' has 1:1 relationship with 'variable'. As 1 variable can be mapped to n aes it's not clear, why relationship between 'ordering option'  and aes is also 1:1 .

If the only purpose of aes here is to define if it's 'positional' than just set such flag while constructing the 'ordering option' .

If aes is necessary then define the aes when 'ordering option' is created from configs using 
`Option.Mapping.toAes(aesName)` .

**Review by alshan — COMMENTED** [review]
Pls name all classes, members and variables using same prefix: either "orderingXXX" or "orderXXX".

File: plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/DataFrame.kt [hunk 2]
```diff
@@ -228,7 +228,7 @@ class DataFrame private constructor(builder: Builder) {
                 .groupBy ({ (value) -> value }) { (_, statCount) -> statCount }
```
[the remark this answers is not in the document]
Comment (OLarionova-HORIS) [reply plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/DataFrame.kt:228]: This code sorts by sum of stat count for the variable,
to get plot with ordering the heights of the geom_bar:
Screenshot 2021-07-06 at 16 53 04

File: plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/data/DataProcessing.kt (hunk 1 above)
Comment (OLarionova-HORIS) [reply plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/data/DataProcessing.kt]: Changed to using of `Option.Mapping.toAes`. 
Also in `TooltipConfig.kt` - similar function was used there.

File: plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/DataFrame.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (alshan) [reply plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/DataFrame.kt:228]: To be discussed @IKupriyanov-HORIS

File: plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/data/DataProcessing.kt (hunk 1 above)
Comment (alshan) [reply plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/data/DataProcessing.kt]: Still not clear what is role of AES here - to be discussed @IKupriyanov-HORIS

**Pull request closed without merging**